### PR TITLE
cli/sql: announce the target tenant in the output

### LINKED ISSUE #93129: cli/sql: clearly announce the target tenant in the output
We want to visually separate SQL shells opened on the system tenant from other tenants, as well as separating secondary tenants from each other.

Some ideas:

- display the target  tenant in a banner when the session is opened
- show the target tenant name/ID in the prompt.
- color the prompt with a highlight if using the system tenant

Jira issue: CRDB-22178

Epic CRDB-23559

### PR DESCRIPTION
**Pull request by knz**
Fixes #93129.
Epic: CRDB-23559

See individual commits for details.

### CODE REVIEW

File: pkg/cli/interactive_tests/test_demo_cli_integration.tcl.disabled [hunk 1]
```diff
@@ -47,28 +47,37 @@ eexpect "does not exist"
 eexpect "defaultdb>"
 send "\\q\r"
 eexpect eof
+
+spawn $argv sql --no-line-editor -u root -d cluster:demo-tenant
```
Comment (rafiss) [inline pkg/cli/interactive_tests/test_demo_cli_integration.tcl.disabled:51]: it seems like this test file is being skipped. should we re-enable it?

there's a lot of other disabled ones too. perhaps we can check that they don't regress with this change, even if they are not re-enabled right now.

**Review by rafiss — APPROVED** [review]

File: pkg/cli/interactive_tests/test_demo_cli_integration.tcl.disabled (hunk 1 above)
Comment (knz) [reply pkg/cli/interactive_tests/test_demo_cli_integration.tcl.disabled:51]: Good call. I'll check them manually in this PR, then send another PR separately to re-enable them.

**Comment by knz** [comment]
I have checked manually that all the disabled tests still pass. I'll send a separate PR to un-skip them.

bors r=rafiss

**Comment by knz** [comment]
lint error

bors r-

**Comment by knz** [comment]
bors r=rafiss

**Pull request merged**